### DISCUSSION TITLE
🎨 Dashboard Redesign

### DISCUSSION THREAD

**Discussion by baylorhenshaw** [coollabsio/coolify#2508] [Feature Requests]
# 🎨 Dashboard Redesign
I've started to use Coolify over the past week, and I really love it! I've only encountered one recurring problem, the dashboard navigation is terrible. That's why I decided to mockup a possible direction you could take the dashboard.
### ⚠️ Current Dashboard Issues
- You need to navigate between 2-3 pages before actually arriving at the configuration for any given service.
- The dashboard barely displays any information, and barely serves a purpose apart from being a projects page.
- You can't take any actions on the dashboard. You need to navigate multiple pages before actually being able to take an action (start, stop, restart, etc.)
### ✅ Possible Solutions
- List all services/containers under a project (dropdown) and add actions for each service (start, stop, restart, etc.)
- Condense environments page into the dashboard page. Instead of having a dedicated page, allow the user to create new environments from the dashboard
- Attempt to remove the extra pages when you can condense a lot of information and basic settings in the dashboard
### 💬 Additional Ideas
- Adding server stats information to the dashboard as mentioned in #2329 and #2288
- Search through projects as mentioned by [Ibnu Rasikh on Discord](https://discord.com/channels/459365938081431553/960818238244134962/1250316057357586514)
- Possibly use a structure similar to the resources tab in the server section.
- Navigate directly to a deployment's website from the dashboard
### ✨ Mockup
I created a quick mockup to see what a new dashboard could look like. This was done really quickly, but it is just so you can envision the difference it could make.

![coolify-reimagined](https://github.com/coollabsio/coolify/assets/64096464/29fb68b2-2836-4627-82b9-7311c7155e88)

**Comment by volfadar** [comment]
also currently projects only sorted alphabetically, i believe "sort by" feature will be very helpful.

**Comment by vai** [comment]
could be interesting. A set of columns for the environments would also help. (look to octopus deploy dashboards for an example)

**Comment by dawnniie** [comment]
Yep, dashboard doesn’t give much info at the moment so something like this would be amazing. +1 for basic stats for servers too

**Comment by AVMG20** [comment]
I would love this. Currently, I have to click three times before I get to the instance screen. This will definitely help while still keeping the same depth.

**Comment by ssoronid** [comment]
I would love to contribute in some substancious QOL for managing this kind of things better / easier.. The thing is, are the Maintainers going to approve / point the PR at this direction? Even if the heavy lift work is lifted by our contributions? @baylorhenshaw 

Would like to redesign and create some features to exnted how deployments works now:

![image](https://github.com/user-attachments/assets/da96efdb-4d56-4564-82a5-ac328705330c)

Into something like this, featuring instant rollbacks, and better identification on whats happening

![image](https://github.com/user-attachments/assets/63607c6a-d584-4583-aef6-ce84bde67f0a)

**Reply by baylorhenshaw** [reply]
This was more of a feature request for the maintainers that points out some of the big UI/UX issues with Coolify. Unfortunately, I don't have time to contribute to actually making this reality. Hopefully the maintainers decide that these ideas are worth looking at, but I haven't heard anything from them.

**Comment by Pedromigacz** [comment]
If this ideas goes forward and you are open to more people involved on the redesign, I'd want to contribute. There are many interaction flow that could be simplified in my opinion.

**Comment by bobbymannino** [comment]
for someone who lives on a smaller screen and has a lot of projects/services i would love to use a search bar, a small popup modal invoked by "cmd-k" would be such a timer saver for me!

**Comment by krizh-p** [comment]
Any updates/plans for improvements to the Shared Variables screen?

https://github.com/coollabsio/coolify/discussions/3643

**Comment by SlickMoose** [comment]
prompting for @baylorhenshaw 's idea to display status of services and health checks in dashboard view

**Comment by woutrbe** [comment]
I find the new dropdown menus in the breadcrumbs slightly confusing. The first breadcrumb lets me switch between projects, the second between environments. So I would've expected the third one to be a quick switch between resources, but instead this is resource specific.

To switch between resources, I first need to select the environment, and then drill down into resources.

Personally:
- I don't find it useful to be able to quick-switch between projects. Because when I'm in a specific resource, I've already been through projects > environment > resource
- Most people organize their resources per project anyway. So what would be the benefit of being able to quick-switch between them?
- I'd like to see a quick-switch for resources within the selected environment, without having to select the environment first.
- In fact, the only useful use of the breadcrumbs to me is being able to quickly switch between environments, and between projects within the environment.

<img width="1177" height="96" alt="Screenshot 2026-01-17 at 12 42 59 PM" src="https://github.com/user-attachments/assets/5c6a5539-ab9d-401a-805f-bfd849eef28e" />

**Comment by woutrbe** [comment]
As a suggestion to my previous comment, I just did this as a quick mockup in Claude. Essentially just adding a 4th item that let's you switch between configuration/deployments/logs/terminal.

This way the resource dropdown can just quick switch between resources. And the environment one just between environments.

<img width="947" height="225" alt="Screenshot 2026-01-19 at 11 25 22 AM" src="https://github.com/user-attachments/assets/7ba63e25-03a0-4762-b6ea-5653371d159a" />

**Comment by LunarECL** [comment]
I split the "sort by" part of this out into its own focused feature request, since the full dashboard redesign is a much bigger scope and this one piece is small and self-contained: https://github.com/coollabsio/coolify/discussions/10811

It covers a sort control (name A→Z / Z→A, recently created, recently updated) and a card/list density toggle on both `/` and `/projects`, with no schema or API changes. Planning to open a PR for it against `next`. Linking back here since this thread (and @volfadar's comment) is where the request originally came from.

**Comment by sofianegargouri** [comment]
My personal take on the dashboard is that it should display an overview of what's happening on Coolify.
My wish would be to have:
- Latest deployments
- Server metrics
- Logs of latest user actions
- Overview of the apps and their statuses, while displaying errored apps well

**Comment by ToreAndreWork** [comment]
Who approved the new UI design? It's so difficult to get an overview, not enough contrast so all the elements kind of just blend together.
